File: 2023/02/cpp/p1.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main(){

    freopen("input.txt","r",stdin);
    string line;
    int sum = 0;
    for(int id = 1; id <= 100; id++){
        getline(cin,line);
        line = line.substr(line.find(":")+2);
        bool isValid = true;
        string num;
        int nCube;
        for(int i = 0; i < (int)line.length()-2; i++){
                num="";
                if(isdigit(line[i])){
                    while(isdigit(line[i])){
                        num+=line[i];
                        i++;
                    }
                    nCube = stoi(num);
                    if(line[i+(int)num.length()-1] == 'b' && nCube > 14){
                        isValid = false;
                        break;
                    }
                    if(line[i+(int)num.length()-1] == 'g' && nCube > 13){
                        isValid = false;
                        break;
                    }
                    if(line[i+(int)num.length()-1] == 'r' && nCube > 12){
                        isValid = false;
                        break;
                    }
                }
        }
        if(isValid) sum+=id;
    }

    cout << sum;

    return 0;
}

File: 2023/03/p1.py
inputFile = open("input.txt","r")


engineSchema = []
speciali = "=%+&$#*-/@"
for line in inputFile:
    engineSchema.append(line.strip())

sum=0
for i in range(len(engineSchema)):
    for j in range (len(engineSchema[i])):
        
        if engineSchema[i][j] in speciali:
            y=i-1
            while y <= i+1:
                x=j-1
                line = engineSchema[y]
                while x <= j+1:
                    if (x >= 0 and x <= len(engineSchema[y])) and (y >= 0 and y <= len(engineSchema)):
                        if line[x].isdigit():
                            start=x
                            
                            while line[start].isdigit() and start>=0:
                                start-=1
                            while x < len(line) and line[x].isdigit():
                                x+=1

                            num=int(line[start+1:x])
                            sum+=num
                    x+=1
                y+=1
                            
print(sum)